diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/BooleanExpressionsRule.kt

```diff
@@ -60,22 +68,39 @@ class BooleanExpressionsRule(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     /**
+     * Converts a complex boolean expression into a string representation, mapping each elementary expression to a letter token.
+     * These tokens are collected into [mapOfExpressionToChar].
+     * For example:
+     * ```
+     * (a > 5 && b != 2) -> A & B
+     * (a > 5 || false) -> A | false
+     * (a > 5 || x.foo()) -> A | B
+     * ```
+     *
      * @param node
-     * @param mapOfExpressionToChar
-     * @return corrected string
+     * @param mapOfExpressionToChar a mutable map for expression->token
+     * @return formatted string representation of expression
      */
+    @Suppress("UnsafeCallOnNullableType")
     internal fun formatBooleanExpressionAsString(node: ASTNode, mapOfExpressionToChar: HashMap<String, Char>): String {
+        val (booleanBinaryExpressions, otherBinaryExpressions) = node
+            .findAllDescendantsWithSpecificType(BINARY_EXPRESSION)
+            .partition { it.text.contains("&&") || it.text.contains("||") }
         // `A` character in ASCII
         var characterAsciiCode = 'A'.code
-        node
-            .findAllNodesWithCondition({ it.elementType == BINARY_EXPRESSION })
-            .filterNot { it.text.contains("&&") || it.text.contains("||") }
+        (otherBinaryExpressions +
```

Comment:
Could you please provide brief comments in the code, about which list we create here, multiple calls of `map` `filter` methods hard to keep in mind

Reply:
Sure, added, thx for the reminder